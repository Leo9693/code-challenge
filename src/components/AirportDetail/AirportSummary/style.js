import styled, { keyframes } from "styled-components";
import { fadeInRight } from "react-animations";

export const AnimatedAirportSummary = styled.div`
    animation: ${keyframes`${fadeInRight}`} 0.7s;
`;
