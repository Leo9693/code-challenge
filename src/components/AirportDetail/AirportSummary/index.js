import React from "react";
import { AnimatedP } from "../style";
import { AnimatedAirportSummary } from "./style";
function index(props) {
    const { airportName, airportCode, city } = props.summary;
    return (
        <AnimatedAirportSummary data-test="container">
            <h3>Airport Summary:</h3>
            <AnimatedP data-test="airportName">AirportName: {airportName}</AnimatedP>
            <AnimatedP data-test="airportCode">AirportCode: {airportCode}</AnimatedP>
            <AnimatedP data-test="city">CityName: {city.cityName}</AnimatedP>
        </AnimatedAirportSummary>
    );
}

export default index;
