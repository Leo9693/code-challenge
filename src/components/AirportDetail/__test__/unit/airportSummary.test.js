import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AirportSummary from "../../AirportSummary";
Enzyme.configure({ adapter: new Adapter() });

test("test ariport Types component", () => {
    const wrapper = shallow(
        <AirportSummary
            summary={{
                airportName: "",
                airportCode: "",
                city: ""
            }}
        />
    );
    expect(wrapper.debug()).toMatchSnapshot();
});
