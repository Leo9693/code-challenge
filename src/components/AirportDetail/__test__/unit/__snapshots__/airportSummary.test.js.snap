// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ariport Types component 1`] = `
"<styled.div data-test=\\"container\\">
  <h3>
    Airport Summary:
  </h3>
  <styled.p data-test=\\"airportName\\">
    AirportName: 
  </styled.p>
  <styled.p data-test=\\"airportCode\\">
    AirportCode: 
  </styled.p>
  <styled.p data-test=\\"city\\">
    CityName: 
  </styled.p>
</styled.div>"
`;
